### PR TITLE
Restore vb compat info and add splash/about template info

### LINKED ISSUE #16655: breaking change with core 3.0 and vb.net splashscreen/about forms
## VB.NET WinForms apps ported to .NET Core 3.0/3.1 that use the splash/about forms won't compile

Visual Studio contains VB.NET specific WinForms templates for a `splashscreen` and `about` form. These templates use the use the `My.Application.Info` object (From the [Microsoft.VisualBasic.ApplicationServices breaking change](https://docs.microsoft.com/en-us/dotnet/core/compatibility/visualbasic#types-in-microsoftvisualbasicapplicationservices-namespace-not-available)) which doesn't exist in .NET Core 3.0/3.1. Users should use the `System.Reflection.Assembly` class to look up the values used by these forms.

These templates were removed from VB.NET in .NET Core but code ported from Framework to Core will experience compile issues with these files.

Related to #16392

### Version introduced

.NET Core 3.0

### Old behavior

Splashscreen/About forms compiled and displayed properly.

### New behavior

Won't compile.

### Recommended action

Use the `System.Reflection.Assembly` class to get this information. A straight port of both forms is available here. (Note that this in unoptimized code. The list of attributes should be cached to reduce form load time)

**Splashscreen**

```vb
Imports System.Reflection

Public NotInheritable Class SplashScreen1

    Private Sub SplashScreen1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        'Set up the dialog text at runtime according to the application's assembly information.  

        'TODO: Customize the application's assembly information in the "Application" pane of the project 
        '  properties dialog (under the "Project" menu).

        'Application title
        Dim appTitle As String = Assembly.GetExecutingAssembly().GetCustomAttribute(Of AssemblyTitleAttribute)()?.Title

        If String.IsNullOrEmpty(appTitle) Then
            appTitle = System.IO.Path.GetFileNameWithoutExtension(Assembly.GetExecutingAssembly().GetName().Name)
        End If

        ApplicationTitle.Text = appTitle

        Dim versionValue = Assembly.GetExecutingAssembly().GetName().Version

        'Format the version information using the text set into the Version control at design time as the
        '  formatting string.  This allows for effective localization if desired.
        '  Build and revision information could be included by using the following code and changing the 
        '  Version control's designtime text to "Version {0}.{1:00}.{2}.{3}" or something similar.  See
        '  String.Format() in Help for more information.
        '
        '    Version.Text = System.String.Format(Version.Text, versionValue.Major, versionValue.Minor, versionValue.Build, versionValue.Revision)

        Version.Text = System.String.Format(Version.Text, versionValue.Major, versionValue.Minor)

        'Copyright info
        Copyright.Text = If(Assembly.GetExecutingAssembly().GetCustomAttribute(Of AssemblyCopyrightAttribute)()?.Copyright, "")
    End Sub

End Class
```

**About**

```vb
Imports System.Reflection

Public NotInheritable Class about

    Private Sub about_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ' Set the title of the form.
        Dim applicationTitle As String = Assembly.GetExecutingAssembly().GetCustomAttribute(Of AssemblyTitleAttribute)()?.Title

        If String.IsNullOrEmpty(applicationTitle) Then
            applicationTitle = System.IO.Path.GetFileNameWithoutExtension(Assembly.GetExecutingAssembly().GetName().Name)
        End If

        Me.Text = String.Format("About {0}", applicationTitle)
        ' Initialize all of the text displayed on the About Box.
        ' TODO: Customize the application's assembly information in the "Application" pane of the project 
        '    properties dialog (under the "Project" menu).
        Me.LabelProductName.Text = If(Assembly.GetExecutingAssembly().GetCustomAttribute(Of AssemblyProductAttribute)()?.Product, "")
        Me.LabelVersion.Text = String.Format("Version {0}", Assembly.GetExecutingAssembly().GetName().Version)
        Me.LabelCopyright.Text = If(Assembly.GetExecutingAssembly().GetCustomAttribute(Of AssemblyCopyrightAttribute)()?.Copyright, "")
        Me.LabelCompanyName.Text = If(Assembly.GetExecutingAssembly().GetCustomAttribute(Of AssemblyCompanyAttribute)()?.Company, "")
        Me.TextBoxDescription.Text = If(Assembly.GetExecutingAssembly().GetCustomAttribute(Of AssemblyDescriptionAttribute)()?.Description, "")
    End Sub

    Private Sub OKButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OKButton.Click
        Me.Close()
    End Sub

End Class

```

### Category

- Visual Basic
- Windows Forms

### Affected APIs

Covered by https://docs.microsoft.com/en-us/dotnet/core/compatibility/visualbasic#types-in-microsoftvisualbasicapplicationservices-namespace-not-available

---
#### Issue metadata

* Issue type: breaking-change

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Restored some VB compat information that was erased by accident.
- Added info about about/splashscreen templates from Visual Studio.

Fixes #16655

Please review this @gewarren 😄 Thanks!

### CODE REVIEW

File: includes/core-changes/visualbasic/3.0/microsoft.visualbasic.applicationservices-unavailable.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿### Types in Microsoft.VisualBasic.ApplicationServices namespace not available
+
+The types in the <xref:Microsoft.VisualBasic.ApplicationServices?displayProperty=fullName> namespace are not available.
+
+#### Version introduced
+
+.NET Core 3.0 Preview 8
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/microsoft.visualbasic.applicationservices-unavailable.md:7]:
```suggestion
.NET Core 3.0
```

File: includes/core-changes/visualbasic/3.0/microsoft.visualbasic.applicationservices-unavailable.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#### Change description
+
+The types in the <xref:Microsoft.VisualBasic.ApplicationServices?displayProperty=fullName> namespace were available in some .NET Core 3.0 Preview releases. They are no longer available starting with .NET Core 3.0 Preview 9.
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/microsoft.visualbasic.applicationservices-unavailable.md:11]:
```suggestion
The types in the <xref:Microsoft.VisualBasic.ApplicationServices?displayProperty=fullName> namespace were available in .NET Framework. They are not available in .NET Core 3.0 - 3.1.
```

File: includes/core-changes/visualbasic/3.0/microsoft.visualbasic.devices-unavailable.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿### Types in Microsoft.VisualBasic.Devices namespace not available
+
+The types in the <xref:Microsoft.VisualBasic.Devices?displayProperty=fullName> namespace are not available.
+
+#### Version introduced
+
+.NET Core 3.0 Preview 8
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/microsoft.visualbasic.devices-unavailable.md:7]:
```suggestion
.NET Core 3.0
```

File: includes/core-changes/visualbasic/3.0/microsoft.visualbasic.devices-unavailable.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+#### Change description
+
+The types in the <xref:Microsoft.VisualBasic.Devices?displayProperty=fullName> namespace were available in some .NET Core 3.0 Preview releases,. They are no longer available starting with .NET Core 3.0 Preview 9.
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/microsoft.visualbasic.devices-unavailable.md:11]:
```suggestion
The types in the <xref:Microsoft.VisualBasic.Devices?displayProperty=fullName> namespace were available in .NET Framework. They are not available in .NET Core 3.0 - 3.1.
```

File: includes/core-changes/visualbasic/3.0/microsoft.visualbasic.myservices-unavailable.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+﻿### Types in Microsoft.VisualBasic.MyServices namespace not available
+
+The types in the <xref:Microsoft.VisualBasic.MyServices?displayProperty=fullName> namespace are not available.
+
+#### Version introduced
+
+.NET Core 3.0 Preview 8
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/microsoft.visualbasic.myservices-unavailable.md:7]:
```suggestion
.NET Core 3.0
```

File: includes/core-changes/visualbasic/3.0/microsoft.visualbasic.myservices-unavailable.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+#### Change description
+
+The types in the <xref:Microsoft.VisualBasic.MyServices?displayProperty=fullName> namespace were available in some .NET Core 3.0 Preview releases. They are no longer available starting with .NET Core 3.0 Preview 9.
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/microsoft.visualbasic.myservices-unavailable.md:11]:
```suggestion
The types in the <xref:Microsoft.VisualBasic.MyServices?displayProperty=fullName> namespace were available in .NET Framework. They are not available in .NET Core 3.0 - 3.1.
```

File: includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
+﻿### About and SplashScreen templates are broken
+
+The `About.vb` and `SplashScreen.vb` files generated by Visual Studio contain references to types in the `My` namespace not available .NET Core 3.0 and 3.1.
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md:3]:
```suggestion
The `About.vb` and `SplashScreen.vb` files generated by Visual Studio contain references to types in the `My` namespace that aren't available .NET Core 3.0 and 3.1.
```

File: includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+#### Version introduced
+
+3.0
+
+#### Change description
+
+.NET Core 3.0 and 3.1 doesn't contain full Visual Basic `My` support. The **About** and **SplashScreen** form templates in Visual Studio for Visual Basic Windows Forms apps reference properties in the `My.Application.Info` type that aren't available.
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md:11]:
```suggestion
.NET Core 3.0 and 3.1 don't contain full Visual Basic `My` support. The **About** and **SplashScreen** form templates in Visual Studio for Visual Basic Windows Forms apps reference properties in the `My.Application.Info` type that aren't available.
```

File: includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+#### Recommended action
+
+Upgrade your app to .NET 5 or later.
+
+-or-
+
+Fix the compiler errors in the **About** and **SplashScreen** types in your app. Use the `System.Reflection.Assembly` class to get the information provided byt he `My.Application.Info` type. A straight port of both forms is available here.
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md:19]:
```suggestion
Fix the compiler errors in the **About** and **SplashScreen** types in your app. Use the `System.Reflection.Assembly` class to get the information provided by the `My.Application.Info` type. A straight port of both forms is available here.
```

File: includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+> [!TIP]
+> This is sample code and unoptimized. The list of attributes should be cached to reduce form load time.
+
+**About**
+
+```vb
+Imports System.Reflection
+
+Public NotInheritable Class about
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md:29]:
```suggestion
Public NotInheritable Class About
```

File: includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+End Class
+```
+
+**SplashScreen**
+
+```vb
+Imports System.Reflection
+
+Public NotInheritable Class SplashScreen1
```
Comment (gewarren) [inline includes/core-changes/visualbasic/3.0/vb-winforms-splash-about-broken.md:62]:
```suggestion
Public NotInheritable Class SplashScreen
```

**Comment by adegeo** [comment]
@gewarren OK accepted the changes

**Comment by gewarren** [comment]
One last thing - we should add those breaking changes to this file: https://docs.microsoft.com/en-us/dotnet/core/compatibility/fx-core

**Comment by adegeo** [comment]
@gewarren I added it to the fx and winforms files

**Comment by gewarren** [comment]
> @gewarren I added it to the fx and winforms files

Should the restored ones also be added to the Fx file? I know they weren't there originally, but seems like they should be, right?

**Comment by adegeo** [comment]
@gewarren ok added those!

**Review by gewarren — APPROVED** [review]

**Pull request merged**